lib/oauth2/client.rb

```diff
@@ -167,10 +167,9 @@ def get_token(params, access_token_opts = {}, extract_access_token = nil, &block
 
       request_opts = {
         raise_errors: options[:raise_errors],
-        parse: params.delete(:parse),
-        snaky: params.delete(:snaky),
+        parse: params.fetch(:parse, Response::DEFAULT_OPTIONS[:parse]),
```

Comment:
This change introduces a bug: by no longer deleting the `parse` or `snaky` parameters from the `params` hash, it results in them getting sent to the oauth server as part of the request body.